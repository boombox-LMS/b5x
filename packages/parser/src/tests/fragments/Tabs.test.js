const {
  exampleTabsMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/Tabs");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Tabs fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleTabsMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed tabs fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed tabs fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
