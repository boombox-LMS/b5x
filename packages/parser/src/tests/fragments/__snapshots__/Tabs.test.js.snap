// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tabs fragment builds as expected The parsed tabs fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
    "test-fragment-uri",
    "test-fragment-uri",
  ],
  "contentType": "Tabs",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "2a285ab348be9248219d8d0d8a27e14e",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
